internal/driver/custommetadata.go

```diff
@@ -0,0 +1,84 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+// setCustomMetadata will return a map containing the fields provided in the call to the function
+func (onvifClient *OnvifClient) setCustomMetadata(device contract.Device, data []byte) (contract.Device, errors.EdgeX) {
+	var dataObj contract.ProtocolProperties
+
+	err := json.Unmarshal(data, &dataObj)
+	if err != nil {
+		return device, errors.NewCommonEdgeX(errors.KindServerError, "failed to unmarshal the json request body", err)
+	}
+	if len(dataObj) == 0 {
+		return device, errors.NewCommonEdgeX(errors.KindServerError, "no data in PUT command", err)
+	}
+	for key, value := range dataObj {
+		trimValue := strings.TrimSpace(value)
+		trimKey := strings.TrimSpace(key)
```

Comment:
if never using the non-trimmed versions, just overwrite them with the same name
```suggestion
		value = strings.TrimSpace(value)
		key = strings.TrimSpace(key)
```